src/resolve/resolved-dns-scope.c

```diff
@@ -778,6 +778,11 @@ DnsScopeMatch dns_scope_good_domain(
                 if (!dns_scope_is_default_route(s))
                         return DNS_SCOPE_NO;
 
+                /* Prefer suitable per-link scopes where possible */
+                DnsServer *server = dns_scope_get_dns_server(s);
+                if (dns_server_is_fallback(server))
```

Comment:
Not a matter, but `dns_server_is_fallback(dns_scope_get_dns_server(s))` is enough.